### PR TITLE
linux: merge files back into single file

### PR DESCRIPTION
**Pull request by bnoordhuis**
I split up linux.c around 2012 for no real reason and now I'm merging
it back together, again for no real reason.

I half-jest. I like the idea of having everything together because I
practically forgot linux-inotify.c existed. It also makes io_uring
experiments a little easier.

Last but not least, it removes about 100 lines of license boilerplate.

<hr>

The first commit is #3777.

### CODE REVIEW

File: src/unix/random-getrandom.c [hunk 1]
```diff
@@ -24,8 +24,6 @@
 
 #ifdef __linux__
 
-#include "linux-syscalls.h"
```
Comment (bnoordhuis) [inline src/unix/random-getrandom.c:27]: The include wasn't necessary because internal.h already pulls everything in.

**Review by santigimeno — APPROVED** [review]

**Review by santigimeno — APPROVED** [review]
LGTM. Looking forward to seeing that io_uring code.

**Comment by bnoordhuis** [comment]
> Looking forward to seeing that io_uring code.

I was going to batch epoll_ctl calls with the IORING_OP_EPOLL_CTL op but that's been deprecated in Linux 6.0. 🤦

Maybe I'll ask on the io-uring mailing list about un-deprecating it if the performance speedups are good enough.

**Comment by zbjornson** [comment]
> It also makes io_uring experiments a little easier.

@bnoordhuis I was just about to see if I can revive #2322, but are you actively working on a replacement for it?

**Comment by bnoordhuis** [comment]
@zbjornson No definite roadmap or timeline but yes.

**Comment by zbjornson** [comment]
@bnoordhuis thanks for the fast reply. Would it be any help if I rebased that PR and land at least the `backend_fd` part, or are you taking a different approach?

**Comment by bnoordhuis** [comment]
Different approach. For starters, I really don't want to take on liburing as a dependency. :-)

**Pull request merged**